### DISCUSSION TITLE
[ENH] Is there a method  for unsupervised classification for small time point?

### DISCUSSION THREAD

**Discussion by wanglu2014** [alan-turing-institute/sktime#1802] [Q&A] [answered]
I searched this github package, and found no recommendation for this kind of problem. My input dataset is a table with 500 samples, 200 features and 3 time points. I need to do unsupervised classification for those samples into two classes. Which method do you recommend?
Thank you for your attention.

**Comment by fkiraly** [comment]
What you mean is probably clustering? Some communities (e.g., GIS, spatial statistics) refer to "clustering", or, more specifically, clustering algorithms applied with the purpose to produce cluster annotations (in "transformer" mode) as "unsupervised classification". The clustering algorithms in the `sktime.clustering` module should allow for that (still a little bit experimental).

Further, since you seem to have a small number of time points, you could also try flattening and using `scikit-learn` directly. The benefit of time series methods materializes usually with more time points. You should try & compare, although with an unsupervised problem you will have to define what your measure of success is.

**Comment by fkiraly** [comment]
converted to discussion, since it seemed like a question rather than a feature request.

FYI @TonyBagnall, since this seems to be about clustering.

**Comment by chrisholder** [comment, marked as answer]
Hi @wanglu2014 thanks for the question! As @fkiraly said it depends on the type of data. By the sounds of it you're describing a time series multivariate clustering problem. Something like our K-means clustering algorithm (find in sktime.clustering). Alternative algorithms which can perform better on some types of data is K-shapes which we don't have an implementation for in sktime but something like ts-learn does - https://tslearn.readthedocs.io/en/stable/gen_modules/clustering/tslearn.clustering.KShape.html

**Comment by wanglu2014** [comment]
Thank you for your timely reply!
